urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlab/examples/Vex.java

```diff
@@ -0,0 +1,223 @@
+package permlab.examples;
+
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Collection;
+
+public class Vex {
+
+    static int BOUND = 3;
+    static final int MAX_LENGTH = 16;
+    static final int[] max3s = new int[MAX_LENGTH];
+    static final int[] largeValues = new int[MAX_LENGTH];
+    static final int[] smallValues = new int[MAX_LENGTH];
+    static final ArrayDeque<int[]> aq = new ArrayDeque<int[]>();
+    static final ArrayDeque<int[]> arrayChildren = new ArrayDeque<int[]>();
+    static int largeStackTop = -1;
+    static int smallStackTop = -1;
+
+    public static boolean avoids2143(int[] values) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 20-72 into a separate method.